openllm-core/src/openllm_core/prompt/prompt_template.py

```diff
@@ -0,0 +1,42 @@
+from __future__ import annotations
+import typing as t
+
+import attr
+
+from .utils import PromptFormatter
+
+default_formatter = PromptFormatter()
+
+@attr.define(slots=True)
+class PromptTemplate:
+  _template: str | None
+  _system_message: str | None
+
+  def __init__(self, template: str | None = None, system_message: str | None = None) -> None:
+    self._template = template
+    self._system_message = system_message
```

Comment:
The whole point of attrs is that you don't have to do this :)

Reply:
Oh I see. Got it, thanks! 🙏